docs/architecture/different_tool_versions.md

```diff
@@ -0,0 +1,130 @@
+Let's look at some concrete examples:
+
+* For the [Kubernetes transport abstraction](../../transport/kubernetes/README.md), the container image to be launched is defined in a (static) configuration.
+  The mechanism to determine the target container image can be extended to take other properties into account, so that the image name and/or tag are generated dynamically.
+  For instance, for a project that sticks with Java 11, the Analyzer endpoint could launch a container based on the image `ort-server-analyzer:java11-latest`, while a project requiring Java 17 would use the image `ort-server-analyzer:java17-latest`.
+  In this scenario, no changes on the ORT Server deployment would be required to add support for new tool versions; only a new container image following the naming conventions in place has to be added to the container registry.
+* Transport implementations based on messaging could similarly derive the name of the target message queue from the version constraints of the current project.
+  In this scenario, container images with specific tool versions are listening on dedicated messages queues and are processing only compatible projects.
+  Supporting a new tool version would mean adding a new message queue and deploying a corresponding listener service.
+
+After the basic idea has been presented, the next section discusses how the specification of tool version constraints and the corresponding dispatching could look like.
+
+## Configuration
+To make sure that projects to be analyzed are processed by workers offering the tools in the required versions, two requirements must be fulfilled:
+
+* Projects must be able to specify version constraints.
+* The transport implementation must be passed such constraints and understand them.
+
+Because of the desired flexibility of ORT Server with regard to its runtime environment, defining strict mapping rules is not possible.
+Users of ORT Server will have their own naming conventions for their container images.
+They will typically require only a limited variance in versions of specific tools, which will also impact the names of container images that are chosen.
+Being fully flexible in the versions of all supported tools would lead to a combinatorics explosion.
+So, the configuration mechanism needs to be quite flexible as well.
+
+When triggering an ORT run, users must be able to specify information about version constraints in a generic way.
+With labels, ORT (and ORT Server) already supports a mechanism to provide arbitrary data that is available during the whole run.
+So it would be possible to define a specific prefix, e.g., _transport:_, for labels that are intended to be evaluated by transport implementations.
+
+When invoking a worker, the Orchestrator could extract all labels starting with this prefix and add them to the message that triggers the worker.
+For this, the `Message` class would have to be extended.
+
+NOTE:
```

Comment:
Should this use `:::note` syntax, while at it?

Reply:
I have removed the "NOTE" altogether, the paragraph does not really stand out to me.